model/Hierarchy.php

```diff
@@ -1,9 +1,13 @@
 <?php
 /**
- * DataObjects that use the Hierarchy extension can be be organised as a hierarchy, with children and parents.
- * The most obvious example of this is SiteTree.
+ * DataObjects that use the Hierarchy extension can be be organised as a hierarchy, with children and parents. The most
+ * obvious example of this is SiteTree.
+ *
  * @package framework
  * @subpackage model
+ *
+ * @property int ParentID
+ * @method DataObject Parent
```

Comment:
I'm guessing this is because `DataExtension::get_extra_config()` defines a relationship, which will allow `Parent()` to be called on `DataObject`'s extended with this `DataExtension`.

It's true that this `DataExtension` will make that method possible, but it doesn't have this method [unless I'm missing a `__get()`/`__call()` implementation in the 3 classes]. It won't be much help for static analysis because this method is inferred through `Object::__call()`, which is hinted as mixed.

What I'm trying to say is this seems like the right thing to do but it doesn't appear to have any effect on static analysis or documentation [for this class in isolation].

I might be missing something so please feel free to set me straight!